### PR TITLE
WIP: Yml migrate

### PR DESCRIPTION
**Pull request by shedninja6**
### Summary
- **apply**: added flag for fogg.yml and changed command line descriptions to match
- **aws_config**: Added flag for fogg.yml
- **config**: remove .json code and currently only supports yaml file **will need guidance on how to add json and yaml support**
- Created an integration test for YAML file. The test is a copy of the original .json integration test
- fogg.yml to testdata, in order to pass tests
- Made minor changes to comments, so that they show the support of .json and .yml file extensions

### Test Plan
Currently failing plan_tests because the tests rely on fogg.json file structure. Currently adding support for fogg.yml

### CODE REVIEW

File: apply/golden_file_test.go [hunk 1]
```diff
@@ -119,100 +119,100 @@ func TestIntegration(t *testing.T) {
 	}
 }
 
-func TestIntegrationYaml(t *testing.T) {
```
Comment (ryanking) [inline apply/golden_file_test.go]: You shouldn't need to create a new go test, juts new test directories in `testdata`

File: cmd/exp/aws_config.go [hunk 2]
```diff
@@ -18,11 +18,10 @@ import (
 	"github.com/spf13/cobra"
 )
 
-//TODO: Potentially revert change
+//TODO: Enable aws_config.go to update yaml and json files
```
Comment (ryanking) [inline cmd/exp/aws_config.go]: this command should only have to read the files, right?

File: cmd/plan.go [hunk 3]
```diff
@@ -10,15 +10,14 @@ import (
...
-	planCmd.Flags().StringP("config", "c", "fogg.json", "Use this to override the fogg config file.")
 	planCmd.Flags().StringP("config", "c", "fogg.yml", "Use this to override the fogg config file.")
 	rootCmd.AddCommand(planCmd)
 }
 
 var planCmd = &cobra.Command{
 	Use:           "plan",
 	Short:         "Run a plan",
-	Long:          "plan will read fogg.yml or fogg.json, use that to generate a plan and print that plan out. It will make no changes.",
+	Long:          "plan will read fogg.yml, use that to generate a plan and print that plan out. It will make no changes.",
```
Comment (ryanking) [inline cmd/plan.go]: I think for the time being we should support both.

File: cmd/upgrade.go [hunk 4]
```diff
@@ -10,7 +10,6 @@ import (
 )
 
 func init() {
-	upgrade.Flags().StringP("config", "c", "fogg.json", "Use this to override the fogg config file.")
```
Comment (ryanking) [inline cmd/upgrade.go]: Even when we are yaml-only we should still allow people to specify a config file.

File: config/v1/config_test.go [hunk 5]
```diff
@@ -83,7 +94,7 @@ func TestParse(t *testing.T) {
 	assert.NotNil(t, c.Modules)
 }
 
-func TestJsonFailure(t *testing.T) {
+func TestYamlFailure(t *testing.T) {
```
Comment (ryanking) [inline config/v1/config_test.go]: rather than replace json, let's do both for now

Comment (shedninja6) [reply config/v1/config_test.go]: If I want to test both I need to make a separate ReadConfig method inside config.go. Is that fine or is there a better way of going around that? @ryanking

**Pull request closed without merging**